src/drivers/camera_trigger/camera_trigger.cpp

```diff
@@ -331,6 +373,51 @@ CameraTrigger::cycle_trampoline(void *arg)
 }
 
 void
+CameraTrigger::distance_trampoline(void *arg)
+{
+
+	CameraTrigger *trig = reinterpret_cast<CameraTrigger *>(arg);
+
+	int poll_interval_usec = 10000;
+
+	// Set trigger based on covered distance
+	if (trig->_vlposition_sub < 0) {
+		trig->_vlposition_sub = orb_subscribe(ORB_ID(vehicle_local_position));
+	}
+
+	struct vehicle_local_position_s pos;
+
+	orb_copy(ORB_ID(vehicle_local_position), trig->_vlposition_sub, &pos);
+
+	if (pos.xy_valid) {
+
+		// Initialize position if not done yet
+		math::Vector<2> current_position(pos.x, pos.y);
+
+		if (!trig->_valid_position) {
+			trig->_last_shoot_position = current_position;
+			trig->_valid_position = pos.xy_valid;
+		}
+
+		float time_now = static_cast<float>(hrt_absolute_time()) / 1.0e6f;
+
+		// Check that distance threshold is exceeded and the time between last shot is large enough
+		if ((trig->_last_shoot_position - current_position).length() >= trig->_distance &&
+		    time_now > trig->_last_shoot_time + trig->_interval / 1000.0f) {
```

Comment:
Why is the interval also tested here? It can be very confusing for the user if he has a very fast vehicle and the trigger is not enabled.